### PR TITLE
[transform] add transform-based fusion strategy.

### PR DESCRIPTION
**Pull request by wyzero**
It's the first step to support transform dialect in BladeDISC.

### CODE REVIEW

File: tao_compiler/mlir/disc/transforms/fusion_utils_transform_based.cc [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  if (lhsTy.getRank() != 2 || rhsTy.getRank() != 2 || outTy.getRank() != 2)
+    return false;
+
+  auto dimNumbers = dotOp.getDotDimensionNumbers();
+  if (dimNumbers.getLhsBatchingDimensions().size() != 0 ||
+      dimNumbers.getRhsBatchingDimensions().size() != 0)
+    return false;
+
+  // TODO(wyzero): support dot_general with fused transpose
+  auto lhsCntractingDims = dimNumbers.getLhsContractingDimensions();
```
Comment (qiuxiafei) [inline tao_compiler/mlir/disc/transforms/fusion_utils_transform_based.cc:42]: nit: lhsC**on**tractingDims?

**Review by qiuxiafei — APPROVED** [review]
LGTM

File: tao_compiler/mlir/disc/transforms/fusion_utils_transform_based.cc (hunk 1 above)
Comment (wyzero) [reply tao_compiler/mlir/disc/transforms/fusion_utils_transform_based.cc:42]: I'll change it in the next PR.

**Comment by wyzero** [comment]
to #787

**Pull request merged**